internal/collector/postgres_metrics.go

```diff
@@ -38,44 +40,100 @@ var ltPG16 json.RawMessage
 
 func EnablePostgresMetrics(ctx context.Context, inCluster *v1beta1.PostgresCluster, config *Config) {
 	if feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
+		log := logging.FromContext(ctx)
+		var err error
+
 		// We must create a copy of the fiveSecondMetrics variable, otherwise we
 		// will continually append to it and blow up our ConfigMap
 		fiveSecondMetricsClone := slices.Clone(fiveSecondMetrics)
+		fiveMinuteMetricsClone := slices.Clone(fiveMinuteMetrics)
 
 		if inCluster.Spec.PostgresVersion >= 17 {
-			fiveSecondMetricsClone, _ = appendToJSONArray(fiveSecondMetricsClone, gtePG17)
+			fiveSecondMetricsClone, err = appendToJSONArray(fiveSecondMetricsClone, gtePG17)
 		} else {
-			fiveSecondMetricsClone, _ = appendToJSONArray(fiveSecondMetricsClone, ltPG17)
+			fiveSecondMetricsClone, err = appendToJSONArray(fiveSecondMetricsClone, ltPG17)
+		}
+		if err != nil {
+			log.Error(err, "error compiling postgres metrics")
 		}
 
 		if inCluster.Spec.PostgresVersion >= 16 {
-			fiveSecondMetricsClone, _ = appendToJSONArray(fiveSecondMetricsClone, gtePG16)
+			fiveSecondMetricsClone, err = appendToJSONArray(fiveSecondMetricsClone, gtePG16)
 		} else {
-			fiveSecondMetricsClone, _ = appendToJSONArray(fiveSecondMetricsClone, ltPG16)
+			fiveSecondMetricsClone, err = appendToJSONArray(fiveSecondMetricsClone, ltPG16)
+		}
+		if err != nil {
+			log.Error(err, "error compiling postgres metrics")
+		}
+
+		// Remove any queries that user has specified in the spec
+		if inCluster.Spec.Instrumentation != nil &&
+			inCluster.Spec.Instrumentation.Metrics != nil &&
+			inCluster.Spec.Instrumentation.Metrics.CustomQueries != nil &&
+			inCluster.Spec.Instrumentation.Metrics.CustomQueries.Remove != nil {
+
+			// Convert json to array of maps
```

Comment:
We start with a yaml for ease of reading; then we generate a json for ease of putting into required OTel; then we unmarshal into a golang map for ease of removing _if_ there's anything to remove.

I don't have any particular ask here, but this feels like a place where we might want to rethink our formats.

Reply:
I agree, it does feel like a lot of conversion, but I don't see any better approach.

Reply:
Oh, but it seems cleaner to me now that we have a struct